src/basic/utf8.c

```diff
@@ -42,7 +42,7 @@ bool unichar_is_valid(char32_t ch) {
                 return false;
         if ((ch >= 0xFDD0) && (ch <= 0xFDEF)) /* Reserved */
                 return false;
-        if ((ch & 0xFFFE) == 0xFFFE) /* BOM (Byte Order Mark) */
+        if (FLAGS_SET(ch, 0xFFFE)) /* BOM (Byte Order Mark) */
```

Comment:
hmm, this is a bit misleading I think. `FLAGS_SET()` suggests in its name that we check flags here, but that's not the case. we check a non-flag value here, masking off some upper and lower bits... I think this should stay as it was.